apps/src/blockly/addons/contextMenu.ts

```diff
@@ -404,3 +471,35 @@ function unregisterDefaultOptions() {
     GoogleBlockly.ContextMenuRegistry.registry.unregister('workspaceDelete');
   } catch (error) {}
 }
+
+function registerCustomBlockOptions() {
+  // Expected registered block options (from Core and plugins):
+  // 0: blockCopyToStorage
+  // 2: blockComment
+  // 5: blockDisable
+  // 6: blockDelete
+  let nextWeight = 7;
+
+  // Custom block options
+  registerHelp(nextWeight++);
+  registerDeletable(nextWeight++);
+  registerMovable(nextWeight++);
+  registerEditable(nextWeight++);
+  registerShadow(nextWeight++);
+  registerUnshadow(nextWeight);
+}
+
+function registerCustomWorkspaceOptions() {
+  // Expected registered workspace options (from Core and plugins):
+  // 0: blockPasteFromStorage
+  // 1: undoWorkspace
+  // 2: redoWorkspace
+  let nextWeight = 3;
+
+  // Custom workspace options
+  registerKeyboardNavigation(nextWeight++);
+  registerAllCursors(nextWeight, NAVIGATION_CURSORS);
+  nextWeight += NAVIGATION_CURSORS.length;
+  registerDarkMode(nextWeight++);
+  registerThemes(nextWeight, themes);
+}
```

Comment:
@ebeastlake I moved things out into two scope-specific functions here (lines 475-505). I'm still using increment operators, but let me know what you think.